### PR TITLE
feat: refactor config, fix typings, add cache safeguards in vector_db, add more logging, rem huggingface

### PR DESCRIPTION
**Pull request by asc11cat**
# Changes Made

Please provide a detailed list of the changes made in this pull request.

1. Centralization of config variables in `sweepai/utils/config.py`
2. Some caching-related safeguards/additions in vector_db
3. Removal of hardcoded constants to make self-hosted deployment saner, including the ability to pass API endpoint/app id for clientside
4. Removal of dead imports
5. Installation of missing chromadb & sentence-transformers deps (lock file)
6. Make 32k gpt-4 model usage optional by setting an env-variable. (Still need to think about how we can reduce the number of tokens to use <32k model)
7. Logging of github app client id/api endpoint in client UI

# Additional Notes

**This PR will work only with modifications to the current secrets in Modals, please see `sweepai/utils/config.py` for that.**

### CODE REVIEW

File: sweepai/core/vector_db.py [hunk 1]
```diff
@@ -274,11 +305,10 @@ def update_index(
...
     installation_id: int,
     sweep_config: SweepConfig = SweepConfig(),
-) -> int:    
-    get_deeplake_vs_from_repo(
-    repo_name,
-    sweep_config,
-    installation_id)
+) -> int:
+    get_deeplake_vs_from_repo(repo_name, installation_id, branch_name=None, sweep_config=sweep_config)
+    # todo: ?
```
Comment (wwzeng1) [inline sweepai/core/vector_db.py:310]: this is good, we run this to hydrate the cache on merge/push

File: sweepai/utils/config.py [hunk 2]
```diff
@@ -35,4 +38,60 @@ def get_branch(repo: Repository) -> str:
...
+# goes under Modal 'github' secret name
+GITHUB_BOT_TOKEN = os.environ.get('GITHUB_BOT_TOKEN')
+GITHUB_APP_ID = os.environ.get('GITHUB_APP_ID')
+GITHUB_BOT_USERNAME = os.environ.get('GITHUB_BOT_USERNAME')
+GITHUB_LABEL_NAME = os.environ.get('GITHUB_LABEL_NAME', 'sweep')
+GITHUB_LABEL_COLOR = os.environ.get('GITHUB_LABEL_COLOR')
+GITHUB_LABEL_DESCRIPTION = os.environ.get('GITHUB_LABEL_DESCRIPTION')
+GITHUB_APP_PEM = os.environ.get('GITHUB_APP_PEM')
+GITHUB_CONFIG_BRANCH = os.environ.get('GITHUB_CONFIG_BRANCH', 'sweep-config')
+GITHUB_DEFAULT_CONFIG = os.environ.get('GITHUB_DEFAULT_CONFIG', """# Reference: https://github.com/sweepai/sweep/blob/main/sweep.yaml
```
Comment (wwzeng1) [inline sweepai/utils/config.py:63]: This can default to none, we only want to change the branch if the user sets it

**Comment by wwzeng1** [comment]
This PR looks great, one of us will merge it to a test branch in sweep, should be done by tomorrow at the latest

**Review by kevinlu1248 — CHANGES_REQUESTED** [review]
Overall a lot of really nice changes. Main complaint is moving constants to config and removing some of the prefixes which would break prod for us :(. Other than that looks pretty good.

**Comment by asc11cat** [comment]
I've adjusted the PR to your comments guys. TLDR:

1. Configs are now under utils/config/server.py OR utils/config/client.py files, and they don't depend on each other.
2. Config vars/envs now have hardcoded fallbacks with values you currently use in your prod env. It should work as it is(?), while still allowing configuration.
3. No chromadb/sentence-transformers deps bloat due to huggingface dead code removal

All other changes in the last commit are purely codestyle.

**Review by kevinlu1248 — APPROVED** [review]
Made some small fixes and merge conflicts. Merging now. Thanks for the contribution!

**Review by kevinlu1248 — APPROVED** [review]
Made some small fixes and merge conflicts. Merging now. Thanks for the contribution!

**Review by kevinlu1248 — APPROVED** [review]
Made some small fixes and merge conflicts. Merging now. Thanks for the contribution!

**Pull request merged**